hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/kafka/SqlAvroTest.java

```diff
@@ -33,6 +33,7 @@
 import io.confluent.kafka.serializers.KafkaAvroSerializer;
 import org.apache.avro.Schema;
 import org.apache.avro.SchemaBuilder;
+import org.apache.avro.generic.GenericRecord;
 import org.apache.avro.generic.GenericRecordBuilder;
 import org.apache.kafka.clients.consumer.ConsumerRecord;
 import org.apache.kafka.clients.consumer.ConsumerRecords;
```

Comment:
when I create kafka mapping with the same key and value schema, only key columns are mapped. According to description in HZ-2754 I'd prefer error in that case.
eg. 
```java
        kafkaMapping(name, ID_SCHEMA, ID_SCHEMA).create();
        sqlService.execute("select get_ddl('relation', '" + name + "')").stream().forEach(System.out::println);
```
gives
```sql
CREATE OR REPLACE EXTERNAL MAPPING "hazelcast"."public"."t_c7b4957b_a354_4fcd_8e27_e8d0d499880b" EXTERNAL NAME "t_c7b4957b_a354_4fcd_8e27_e8d0d499880b" (
  "id" INTEGER EXTERNAL NAME "__key.id"
)
TYPE "Kafka"
OBJECT TYPE "Topic"
OPTIONS (
  'keyFormat'='avro',
  'valueFormat'='avro',
  'bootstrap.servers'='PLAINTEXT://localhost:49229',
  'auto.offset.reset'='earliest',
  'keyAvroSchema'='{"type":"record","name":"sql","namespace":"jet","fields":[{"name":"id","type":["null","int"],"default":null}]}',
  'valueAvroSchema'='{"type":"record","name":"sql","namespace":"jet","fields":[{"name":"id","type":["null","int"],"default":null}]}'
)
```

Reply:
the same should apply to case when we have conflicting fields in avro and non-avro schema. How do we handle such situation in other formats?

Reply:
apparently we also ignore duplicates

Reply:
I see @Fly-Style  also mentioned it

Reply:
Currently, key fields shadow value fields, which is controlled by `KvMetadataResolvers.resolveAndValidateFields(List, Map, NodeEngine)`. I modified it to fail on duplicate resolved columns, but still need to modify the tests that depend on this behavior.

Reply:
as discussed, this could break some users' code, so let's postpone that. Maybe we should add a flag that would restore previous behavior if someone needs that.

Reply:
also, after the change what should happen to existing MAPPINGs if they have the same field in key and value? Probably nothing, but to recreate them users would need give column list explicitly.

Reply:
HZ-3706